### PR TITLE
who even uses rei 2 these days

### PR DESCRIPTION
**Pull request by shedaniel**
don't worry i have used kotlin for 2 hours
i can definitely make quality kotlin code
jk

please check if there is any errors

### CODE REVIEW

**Comment by shedaniel** [comment]
what did you do to make conflicts
fixing them

**Comment by natanfudge** [comment]
ooooo thx

File: src/main/kotlin/spatialcrafting/compat/rei/ReiSpatialCraftingPlugin.kt [hunk 1]
```diff
@@ -11,39 +15,32 @@ import spatialcrafting.recipe.SpatialRecipe
...
-    private fun <T : Recipe<*>> RecipeHelper.registerRecipes(size: Int, recipeFilter: (Recipe<*>) -> Boolean, mappingFunction: (T) -> RecipeDisplay<*>) {
+    private fun <T : Recipe<*>> RecipeHelper.registerRecipes(size: Int, recipeFilter: (Recipe<*>) -> Boolean, mappingFunction: (T) -> RecipeDisplay) {
         registerRecipes(ReiSpatialCraftingCategory.id(size), Function(recipeFilter), Function(mappingFunction))
     }
 
 
     override fun registerRecipeDisplays(recipeHelper: RecipeHelper) {
         for (i in CrafterSizes) {
             recipeHelper.registerRecipes<SpatialRecipe>(i,
-                    recipeFilter = { recipe -> recipe is SpatialRecipe && recipe.minimumCrafterSize == i  },
```
Comment (natanfudge) [inline src/main/kotlin/spatialcrafting/compat/rei/ReiSpatialCraftingPlugin.kt:46]: wow

**Comment by natanfudge** [comment]
Oh right I changed some things

File: src/main/kotlin/spatialcrafting/compat/rei/util/EmptyWidget.kt [hunk 2]
```diff
@@ -4,7 +4,7 @@ import me.shedaniel.rei.gui.widget.Widget
 import net.minecraft.client.gui.Element
 
 class EmptyWidget : Widget() {
-    override fun children(): List<Element> = listOf()
+    override fun children(): List<Element> = emptyList()
```
Comment (natanfudge) [inline src/main/kotlin/spatialcrafting/compat/rei/util/EmptyWidget.kt:7]: I think the implementation of `listOf()` is `emptyList()`

Comment (shedaniel) [reply src/main/kotlin/spatialcrafting/compat/rei/util/EmptyWidget.kt:7]: `emptyList()` returns `EmptyList`, which is immutable & doesn't create a new list instance.

File: src/main/kotlin/spatialcrafting/compat/rei/ReiSpatialCraftingPlugin.kt (hunk 1 above)
Comment (shedaniel) [reply src/main/kotlin/spatialcrafting/compat/rei/ReiSpatialCraftingPlugin.kt:46]: It is a reformatting, look at the line under this

File: src/main/kotlin/spatialcrafting/compat/rei/util/EmptyWidget.kt (hunk 2 above)
Comment (natanfudge) [reply src/main/kotlin/spatialcrafting/compat/rei/util/EmptyWidget.kt:7]: Yes but
image

Comment (shedaniel) [reply src/main/kotlin/spatialcrafting/compat/rei/util/EmptyWidget.kt:7]: I tried

**Pull request merged**